### PR TITLE
0ad A24b building now

### PR DESCRIPTION
**Pull request by zeri42**
Using the Spidermonkey_78.6 that is part of the source bundle.

###### Motivation for this change

New major release by wildfiregames developer team ... playing the game in multiplayer requires to use this version.

###### Things done

As the Spidermonkey version of 0ad is incompatible with that already in nixpkgs we use the version shipped with the 0ad sources.

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/0ad/game.nix [hunk 1]
```diff
@@ -1,30 +1,115 @@
-{ stdenv, lib, perl, fetchurl, python2
-, pkg-config, spidermonkey_38, boost, icu, libxml2, libpng, libsodium
-, libjpeg, zlib, curl, libogg, libvorbis, enet, miniupnpc
-, openal, libGLU, libGL, xorgproto, libX11, libXcursor, nspr, SDL2
-, gloox, nvidia-texture-tools
-, withEditor ? true, wxGTK ? null
+{stdenv
+,cargo
```
Comment (andir) [inline pkgs/games/0ad/game.nix:2]: Commonly we add a space after the comma whenever we use this style for dependencies.

e.g.:

```
{ stdenv
, cargo
, lib
, ...
}
```

File: pkgs/games/0ad/game.nix [hunk 2]
```diff
@@ -1,30 +1,115 @@
...
   src = fetchurl {
-    url = "http://releases.wildfiregames.com/0ad-${version}-alpha-unix-build.tar.xz";
-    sha256 = "0draa53xg69i5qhqym85658m45xhwkbiimaldj4sr3703rjgggq1";
+   url = "http://releases.wildfiregames.com/0ad-${version}-alpha-unix-build.tar.xz";
+   sha256 = "1a1py45hkh2cswi09vbf9chikgxdv9xplsmg6sv6xhdznv4j6p1j";
   };
 
-  nativeBuildInputs = [ python2 perl pkg-config ];
+  #this is required so SpiderMonkey builds with (cargo-link is searching impure paths)
+  NIX_ENFORCE_PURITY=0;
```
Comment (andir) [inline pkgs/games/0ad/game.nix:57]: Please add spaces around the `=`.

File: pkgs/games/0ad/game.nix [hunk 3]
```diff
@@ -1,30 +1,115 @@
...
+   gloox
+   nvidia-texture-tools
+   libsodium
+   fmt
+   libidn
+   cargo
+   rustc
+   python3
+   yasm
+   nspr] ++ lib.optional withEditor wxGTK;
```
Comment (andir) [inline pkgs/games/0ad/game.nix:112]: The closing `]` should be after a linebreak.

The following style is commonly found in nixpkgs:
```nix
myListOfThings = [
  somePackage
] + lib.optinal false package;
```

File: pkgs/games/0ad/game.nix [hunk 4]
```diff
@@ -33,37 +118,24 @@ stdenv.mkDerivation rec {
...
     # Delete shipped libraries which we don't need.
-    rm -rf libraries/source/{enet,miniupnpc,nvtt,spidermonkey}
-
+    rm -rf libraries/source/{enet,miniupnpc,nvtt}
     # Workaround invalid pkg-config name for mozjs
     mkdir pkg-config
-    ln -s ${spidermonkey_38}/lib/pkgconfig/* pkg-config/mozjs-38.pc
     PKG_CONFIG_PATH="$PWD/pkg-config:$PKG_CONFIG_PATH"
-
+    # bash doesn't know die anymore
```
Comment (andir) [inline pkgs/games/0ad/game.nix:127]: Is this an upstream packaging issue that we should report?

**Review by andir — CHANGES_REQUESTED** [review]
Please fix the commit message. It currently doesns't conform to https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md.


There are some syntax nits that should be fixed.

I haven't tried running it yet.

**Comment by zeri42** [comment]
there seems to be someone who has found a solution to use the spindermonkey from nixpkgs ... this is to be prefered I'm closing this PR

**Pull request closed without merging**